src/inventorymanager.cpp

```diff
@@ -154,6 +154,67 @@ IMoveAction::IMoveAction(std::istream &is, bool somewhere) :
 	}
 }
 
+void IMoveAction::swapDirections()
+{
+	std::swap(from_inv, to_inv);
+	std::swap(from_list, to_list);
+	std::swap(from_i, to_i);
+}
+
+void IMoveAction::onPutAndOnTake(ItemStack &src_item, ServerActiveObject* player)
+{
+	if (from_inv.type == InventoryLocation::DETACHED)
+		PLAYER_TO_SA(player)->detached_inventory_OnPut(
```

Comment:
When you're cleaning this up anyway why not make it so `PLAYER_TO_SA(player)` is only called once?

Reply:
Will do.